### DISCUSSION TITLE
npm install 500 internal server error uplink is offline

### DISCUSSION THREAD

**Discussion by nicky132** [verdaccio/verdaccio#3491] [Q&A] [answered]
**Your Environment**
  
 * **verdaccio version**: Docker version 20.10.17, build 100c701
 * **node version** [12.x.x, 14.x.x]:v14.17.5
 * **package manager**: [npm@6, yarn@1]
 * **os**: [CentOS Linux release 7.9.2009 (Core)]
 * **platform**: [docker]

**Describe the bug**
<img width="873" alt="lALPJwY7VJIjAZ_M1M0G0g_1746_212" src="https://user-images.githubusercontent.com/34881461/200994678-4063e665-a1c0-4f3a-937d-3288c95142a1.png">
![1668051707268_59701E21-264C-4a7e-AAC3-D01603A9D029](https://user-images.githubusercontent.com/34881461/200995459-14beb0cd-6531-4b05-a59e-972a1eb332f3.png)

**To Reproduce**

**Expected behavior**

**Screenshots, server logs, package manager log**

**Configuration File (cat ~/.config/verdaccio/config.yaml)**

**Environment information**

**Debugging output**

- `$ NODE_DEBUG=request verdaccio` display request calls (verdaccio <--> uplinks)
- `$ DEBUG=verdaccio* verdaccio` enable extreme verdaccio debug mode (verdaccio api)
- `$ npm -ddd` prints:
- `$ npm config get registry` prints:

**Contribute to Verdaccio**

- [ ] I'm willing to fix this bug 🥇

**Comment by juanpicado** [comment, marked as answer]
That error is normal if there is no connection, you are using docker so might have other issues, that's what I got from the little info you have provided, hope it helps.

**Reply by nicky132** [reply]
thank you for your reply. what other info am i supposed to provide? so you suggest i need to use verdaccio cli?do not use docker verdaccio

**Reply by juanpicado** [reply]
You can use both, if it works with npm should works with docker, there is no diff, if fails in both problem might be in your side.

**Reply by nicky132** [reply]
> You can use both, if it works with npm should works with docker, there is no diff, if fails in both problem might be in your side.

thanks,so there's nothing wrong with the verdaccio config which the error said Internal Server Error uplink is offline ?

storage: /verdaccio/storage
# 存放所有插件的目录
plugins: /verdaccio/plugins

web:
  # 网站Title
  title: "xxxx"
  # 禁用Gravatar头像
  # gravatar: false
  # 排序方式 asc|desc
  # sort_packages: asc
  # 是否启用暗黑模式
  # darkMode: true
  # logo地址
  # logo: http://somedomain/somelogo.png
  # favicon地址
  # favicon: http://somedomain/favicon.ico | /path/favicon.ico

# i18n翻译配置
# i18n:
# 可用列表见：https://github.com/verdaccio/ui/tree/master/i18n/translations
#   web: en-US

publish:
  allow_offline: true
auth:
  htpasswd:
    file: /verdaccio/conf/htpasswd
    # 最大注册用户数，默认为 "+inf".
    # 可用通过设置为-1禁止注册
    # max_users: 1000

# 上游npm库，这里可用设置为淘宝
uplinks:
  taobao:
    url: https://registry.npm.taobao.org
    timeout: 3000ms
  yarnpkg:
    url: https://registry.yarnpkg.com/
    timeout: 6000ms
  npmjs:
    url: https://registry.npmjs.org/
    timeout: 6000ms
  local:
    url: http://10.100.218.67:4873/
    timeout: 6000ms
packages:
  # 作用域包
  '@*/*':  
    # 允许所有人访问
    access: $all
    # 注册用户可访问
    publish: superAdmin admin
    # 注册用户可访问
    unpublish: superAdmin admin
    # 代理 表示没有的仓库会去这个npmjs 里面去找 
    proxy: taobao yarnpkg npmjs local

  '**':
    # 默认情况下所有用户 (包括未授权用户) 都可以查看和发布任意包
    # 你可以指定 用户名/分组名 (取决于你使用什么授权插件，默认的授权插件是内置的 htpasswd)
    # 访问权限有三个关键词: "$all", "$anonymous", "$authenticated"
    # $all 表示不限制，任何人可访问；
    # $anonymous 表示未注册用户可访问；
    # $authenticated 表示只有注册用户可访问
    access: $all

    # 允许所有注册用户发布/撤销已发布的软件包
    # (注意：默认情况下任何人都可以注册)
    publish: superAdmin admin
    unpublish: superAdmin admin

    # 如果package 不存在,就向代理的上游服务发起请求
    proxy: taobao yarnpkg npmjs local

# 您可以指定传入连接的HTTP /1.1服务器保持活动超时（以秒为单位）。
# 值为0会使http服务器的行为类似于8.0.0之前的Node.js版本，后者没有保持活动超时。
# 解决方法：通过给定的配置可以解决以下问题
server:
  keepAliveTimeout: 60
# 中间件
middlewares:
  audit:
    enabled: true

# 日志设置
logs: { type: stdout, format: pretty, level: http }
max_body_size: 100mb # 设置最大包体积
# 开放远程访问，允许所有IP
listen:
  - 10.100.218.67:4873
i18n:
  # 修改默认中文展示
  web: zh-CN

**Comment by juanpicado** [comment]
>  the error said Internal Server Error uplink is offline ?

This error means network failed, you may want check here https://github.com/verdaccio/verdaccio/discussions/3492
